### ISSUE TITLE
is train.py for the VAE implementation ? latent variable z is not sampled from its posterior p(z|x)

### ISSUE THREAD

**Issue by helenlai** [fishmoon1234/DAG-GNN#17]
Hi there, in train.py line 327, it seems like the 'logits' is directly used as the latent variable z. Based on the loss function KL_gaussian_sem it seems like that would be the log of mean of z.

The original VAE formulation requires the z to be sampled from its posterior which then requires the reparameterisation trick to correctly approximate the expectation with monte carlo estimate. Skipping this step would make the resulting implementation deviate quite far from the VAE framework. 

I am also not sure why KL_gaussian_sem is used instead of KL_gaussian, this and the above choices only make sense if you are not assuming the implemented model in train.py won't take the form of a VAE. Could you please confirm this?

**Comment by messcode** [comment]
Hi @helenlai 
I am also confused about this problem. The implementation is different from the description in the paper. I tried to sample the latent variables from the Gaussian distribution, but the inferred causal graph is inaccurate.
I am not sure what's going wrong.

**Comment by Blues-He** [comment]
I am also confused about this question and look forward to the author's answer.
